specification/under_development/profiles/Adaptive_Profile.md

```diff
@@ -0,0 +1,1028 @@
+  ret void
+}
+
+declare void @__quantum__qis__cnot__body(%Qubit*, %Qubit*) local_unnamed_addr
+
+declare void @__quantum__qis__h__body(%Qubit*) local_unnamed_addr
+
+declare void @__quantum__qis__x__body(%Qubit*) local_unnamed_addr
+
+declare void @__quantum__qis__z__body(%Qubit*) local_unnamed_addr
+
+declare void @__quantum__qis__reset__body(%Qubit*) local_unnamed_addr
+
+declare void @__quantum__rt__result_record_output(%Result*, i8*)
+
+declare void @__quantum__qis__mz__body(%Qubit*, %Result*)
+
+declare i1 @__quantum__qis__read_result__body(%Result*)
+
+; attributes
+
+attributes #0 = { "entry_point" "qir_profiles"="adaptive_profile" "output_labeling_schema"="schema_id" "required_num_qubits"="6" "required_num_results"="6" }
+
+; module flags
+
+!llvm.module.flags = !{!0, !1, !2, !3, !4, !5, !6, !7, !8, !9, !10}
+
+!0 = !{i32 1, !"qir_major_version", i32 1}
+!1 = !{i32 7, !"qir_minor_version", i32 0}
+!2 = !{i32 1, !"dynamic_qubit_management", i1 false}
+!3 = !{i32 1, !"dynamic_result_management", i1 false}
+!4 = !{i32 1, !"qubit_resetting", i1 true} ; bullet 8 qubit resetting
+!5 = !{i32 1, !"classical_ints", i1 false} ; ...
+!6 = !{i32 1, !"classical_floats", i1 false}
+!7 = !{i32 1, !"classical_fixed_points", i1 false}
+!8 = !{i32 1, !"user_functions", i1 false}
+!9 = !{i32 1, !"backwards_branching", i1 false}
+!10 = !{i32 1, !"multiple_target_branching", i1 false}
+```
+
+The program performs gate teleportation, and it uses conditional single qubit
+gates and mid-circuit measurements to effect control flow.
+
+For the sake of clarity, the code above does not contain any [debug
+symbols](https://releases.llvm.org/13.0.0/docs/tutorial/MyFirstLanguageFrontend/LangImpl09.html?highlight=debug%20symbols).
+Debug symbols contain information that is used by a debugger to relate failures
+during execution back to the original source code. While we expect this to be
+primarily useful for execution on simulators, debug symbols are both easy to
+ignore and may be useful to generate helpful error messages for compilation
+failures that happen only late in the process. We hence see no reason to
+disallow them from occurring in the bitcode but will not detail their use any
+further. We defer to existing resources for more information about how to
+generate and use debug symbols.
+
+## Entry Point Definition
+
+The bitcode contains the definition of the LLVM function that should be invoked
+when the program is executed, referred to as "entry point" in the rest of this
+profile specification. The name of this function may be chosen freely, as long
+as it is a valid [global
+identifier](https://llvm.org/docs/LangRef.html#identifiers) according to the
+LLVM standard. The entry point is identified by a custom function attribute; the
+section on [attributes](#attributes) defines which attributes must be attached
+to the entry point function.
+
+An entry point function may not take any parameters and must either return void
```

Comment:
I think we should always require returning an integer. The Base Profile requires returning an integer even though there are no error conditions. 
```suggestion
An entry point function may not take any parameters and must either return void
```